src/gt4py/next/program_processors/runners/dace_fieldview/transformations/strides.py

```diff
@@ -426,59 +435,137 @@ def next_edges_by_connector(
+        if len(new_strides) != 0:
+            raise ValueError(f"Dimensional error for '{inner_data}' in '{nsdfg_node.label}'.")
         return
 
-    assert isinstance(inner_desc, dace.data.Array)
+    if not isinstance(inner_desc, dace_data.Array):
+        raise TypeError(
+            f"Expected that '{inner_data}' is an 'Array' but it is '{type(inner_desc).__name__}'."
+        )
+
+    # Now we actually replace the strides, there are two ways of doing it.
+    #  The first is to create an alias in the `symbol_mapping`, however,
+    #  this is only possible if the current strides are singular symbols,
+    #  like `__a_strides_1`, but not expressions such as `horizontal_end - horizontal_start`
+    #  or literal values.
+    #  The second way would be to replace `strides` attributer of the
+    #  inner data descriptor. In case the new stride consists of expressions
+    #  such as `value1 - value2` we have to make them available inside the
+    #  NestedSDFG. However, it could be that the strides is used somewhere else.
+    # We will do the following, if `ignore_symbol_mapping` is `False` and
+    #  the strides of the inner descriptors are symbols, we will use the
+    #  symbol mapping. Otherwise, we will replace the `strides` attribute
+    #  of the inner descriptor, in addition we will install a remapping,
+    #  for those values that were a symbol.
     if (not ignore_symbol_mapping) and all(
-        isinstance(inner_stride, dace.symbol) for inner_stride in inner_desc.strides
+        isinstance(inner_stride, dace.symbol) for inner_stride in inner_strides_init
     ):
+        # Use the symbol
         for inner_stride, outer_stride in zip(inner_desc.strides, new_strides, strict=True):
             nsdfg_node.symbol_mapping[inner_stride.name] = outer_stride
     else:
-        assert len(inner_desc.shape) == len(new_strides)
+        # We have to replace the `strides` attribute of the inner descriptor.
         inner_desc.set_shape(inner_desc.shape, new_strides)
 
-        new_strides_symbols: list[dace.symbol] = functools.reduce(
-            lambda acc, itm: (acc + list(itm.free_symbols))  # type: ignore[union-attr]
-            if dace.symbolic.issymbolic(itm)
-            else acc,
-            new_strides,
-            [],
-        )
-        new_strides_free_symbols = {
-            sym for sym in new_strides_symbols if sym.name not in nsdfg_node.sdfg.symbols
+        # Now find the free symbols that the new strides need.
+        new_strides_symbols: list[str] = []
+        for new_stride_dim in new_strides:
+            if dace.symbolic.issymbolic(new_stride_dim):
+                new_strides_symbols.append(str(new_stride_dim))
+            else:
+                new_strides_symbols.extend(sym for sym in new_stride_dim.free_symbols)
+
+        # Now we determine the set of symbols that should be mapped inside the NestedSDFG.
+        #  We will exclude all that are already inside the `symbol_mapping` (we do not
```

Comment:
I would prefer to have a separate check to ensure that the symbols already inside `symbol_mapping` map to equivalent expressions.

Reply:
I agree.
However, you could apply this check only to the things that is defined in the symbol mapping and there it might be hard to do in general.
Furthermore, you could have interstate assignments.

I added a Todo, if you have concrete ideas let me know.